src/averageAges.js

```diff
@@ -55,10 +76,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const children = people.filter((person, _, peopleArr) => {
+    const hasMother = peopleArr.find(mother => mother.name === person.mother);
+
+    return hasMother && (onlyWithSon ? person.sex === 'm' : true);
+  });
+
+  const childrenTotalAge = children.reduce((acc, child,) => {
+    const mother = people.find(person => person.name === child.mother);
+
+    if (mother !== child.born) {
+      return acc + child.born - mother.born;
+    }
+
+    return acc;
+  }, 0);
+
+  return childrenTotalAge / children.length;
```

Comment:
Similar to previously mentioned cases, handle division by 0 for calculateAverageAgeDiff to handle potentially empty child arrays, averting NaN outcomes.